### PR TITLE
fix: onClick borked in non-read only DeFi opportunities

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Fast-follow of https://github.com/shapeshift/web/pull/7678, which borked onClick handlers for natively supported DeFi opportunities:

https://jam.dev/c/44cff944-f6df-488b-804d-09cb6a2e10d9

## Issue (if applicable)

closes N/A

## Risk

> High Risk PRs Require 2 approvals

> What protocols, transaction types, wallets or contract interactions might be affected by this PR?

Low, immediate fix of an introduced bug

## Testing

- Natively supported DeFi opportunities are clickable and route to the correct place
- Frankenstein opportunities (read: technically read-only as we don't have first-class DeFi modal for them, but we actually support them in Portals) route to swapper
- RFOX still route to RFOX page
- Actual read-only opportunities if held (e.g Idle) route to dApp page

### Engineering

- ^

### Operations

- [ ] :checkered_flag: My feature is behind a flag and doesn't require operations testing (yet)

- ^

## Screenshots (if applicable)

https://jam.dev/c/85a2e92c-6e9c-491c-b35f-f622e464da16

### CODE REVIEW

File: src/components/EarnDashboard/components/ProviderDetails/OpportunityRow.tsx [hunk 1]
```diff
@@ -86,7 +86,7 @@ export const OpportunityRow: React.FC<
 
   const handleClick = useCallback(
     (action: DefiAction) => {
-      if (asset?.isPool) {
+      if (asset?.isPool && opportunity.isReadOnly) {
```
Comment (gomesalexandre) [inline src/components/EarnDashboard/components/ProviderDetails/OpportunityRow.tsx:89]: The actual fix

File: src/components/EarnDashboard/components/ProviderDetails/OpportunityRow.tsx [hunk 2]
```diff
@@ -140,7 +140,7 @@ export const OpportunityRow: React.FC<
             <NestedAsset
               key={rewardAssetId}
               isClaimableRewards={isClaimableRewards}
-              isExternal={opportunity.isReadOnly}
+              isExternal={opportunity.isReadOnly && !asset?.isPool}
```
Comment (gomesalexandre) [inline src/components/EarnDashboard/components/ProviderDetails/OpportunityRow.tsx:143]: Fix to not show the read-only icon for read-only assets that route to swapper for Portals swapping

File: src/components/EarnDashboard/components/ProviderDetails/OpportunityRow.tsx [hunk 3]
```diff
@@ -218,7 +219,7 @@ export const OpportunityRow: React.FC<
             icons={icons}
             subText={assetCellSubText}
             justifyContent='flex-start'
-            isExternal={opportunity.isReadOnly}
+            isExternal={opportunity.isReadOnly && !asset?.isPool}
```
Comment (gomesalexandre) [inline src/components/EarnDashboard/components/ProviderDetails/OpportunityRow.tsx:222]: Fix to not show the read-only icon for read-only assets that route to swapper for Portals swapping

**Review by NeOMakinG — APPROVED** [review]
https://jam.dev/c/8bf0ca37-7bcf-426c-945c-7114f05198ef

Looks good, thanks for taking the time to remove that external icon I suggested in the initial PR 🙏

**Pull request merged**